cmd/event-notifier.go

```diff
@@ -418,23 +424,27 @@ func loadListenerConfig(bucket string, objAPI ObjectLayer) ([]listenerConfig, er
 	var buffer bytes.Buffer
 	err := objAPI.GetObject(minioMetaBucket, lcPath, 0, -1, &buffer)
 	if err != nil {
-		// 'notification.xml' not found return
+		// 'listener.json' not found return
 		// 'errNoSuchNotifications'.  This is default when no
-		// bucket listeners are found on the bucket.
-		if isErrObjectNotFound(err) {
+		// bucket listeners are found on the bucket
+		if isErrObjectNotFound(err) || isErrIncompleteBody(err) {
 			return nil, errNoSuchNotifications
 		}
 		errorIf(err, "Unable to load bucket-listeners for bucket %s", bucket)
 		// Returns error for other errors.
 		return nil, err
 	}
 
-	// Unmarshal notification bytes.
+	// if `listener.json` we should return NoSuchNotifications.
+	if buffer.Len() == 0 {
+		return nil, errNoSuchNotifications
```

Comment:
shouldn't this be `return nil, traceError(errNoSuchNotifications)`?

Reply:
Yes correct..